tests/linearGaussian_npse_test.py

```diff
@@ -179,27 +182,94 @@ def test_npse_iid_inference(num_trials):
     prior_mean = zeros(num_dim)
     prior_cov = eye(num_dim)
     prior = MultivariateNormal(loc=prior_mean, covariance_matrix=prior_cov)
-    gt_posterior = true_posterior_linear_gaussian_mvn_prior(
-        x_o, likelihood_shift, likelihood_cov, prior_mean, prior_cov
-    )
-    target_samples = gt_posterior.sample((num_samples,))
 
-    inference = NPSE(prior, show_progress_bars=True)
+    # This check that our method to handle "general" priors works.
+    # i.e. if NPSE does not get a proper passed by the user.
+    prior_npse = prior if not with_improper_prior else None
```

Comment:
use an explicit `prior` arg for this test and have params `gaussian` and `None`?